code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1610,87 +1610,240 @@ GLOBAL_LIST_EMPTY(roundstart_races)
+ * * humi (required)(type: /mob/living/carbon/human) The mob we will target
  */
-/datum/species/proc/handle_environment(datum/gas_mixture/environment, mob/living/carbon/human/H)
-	var/areatemp = H.get_temperature(environment)
+/datum/species/proc/handle_environment(datum/gas_mixture/environment, mob/living/carbon/human/humi)
+	handle_environment_pressure(environment, humi)
 
-	if(H.stat != DEAD) // If you are dead your body does not stabilize naturally
-		natural_bodytemperature_stabilization(environment, H)
+/**
+ * Body temperature handler for species
+ * vars:
+ * * humi (required)(type: /mob/living/carbon/human) The mob we will target
+ */
+/datum/species/proc/handle_body_temperature(mob/living/carbon/human/humi)
+	if(humi.stat == DEAD) // If you are dead none of this matters
+		return
+
+	body_temperature_core(humi)
+	body_temperature_skin(humi)
+	body_temperature_alerts(humi)
+	body_temperature_damage(humi)
+
+/**
+ * Used to stabilize the core temperature back to normal on living mobs
+ *
+ * The metabolisim heats up the core of the mob trying to keep it at the normal body temp
+ * vars:
+ * * humi (required) The mob we will stabilize
+ */
+/datum/species/proc/body_temperature_core(mob/living/carbon/human/humi)
+	var/natural_change = get_temp_change_amount(humi.get_body_temp_normal() - humi.coretemperature, 0.12)
+	humi.adjust_coretemperature(humi.metabolism_efficiency * natural_change)
+
+/**
+ * Used to normalize the skin temperature on living mobs
+ * vars:
+ * * humi (required) The mob we will targeting
+ */
+/datum/species/proc/body_temperature_skin(mob/living/carbon/human/humi)
+
+	// change the core based on the skin temp
+	var/skin_core_diff = humi.bodytemperature - humi.coretemperature
+	var/skin_core_change = get_temp_change_amount(skin_core_diff, 0.08)
+
+	humi.adjust_coretemperature(skin_core_change)
+
+	// get the enviroment details
+	var/datum/gas_mixture/environment = humi.loc.return_air()
+	if(!environment)
+		return
+
+	var/area_temp = humi.get_temperature(environment)
+	// Get the insulation value based on the area's temp
+	var/thermal_protection = humi.get_insulation_protection(area_temp)
 
-	if(!H.on_fire || areatemp > H.bodytemperature) // If we are not on fire or the area is hotter
-		H.adjust_bodytemperature((areatemp - H.bodytemperature), use_insulation=TRUE, use_steps=TRUE)
+	// Changes to the skin temperature based on the area
+	var/area_skin_diff = area_temp - humi.bodytemperature
+	if(!humi.on_fire || area_skin_diff > 0)
+		var/area_skin_change = get_temp_change_amount(area_skin_diff, 0.1)
 
-/// Handle the body temperature status effects for the species
-/// Traits for resitance to heat or cold are handled here.
-/datum/species/proc/handle_body_temperature(mob/living/carbon/human/H)
+		// Are we sweaty, check against unmodified body temp.
+		if(humi.get_body_temp_normal(apply_change=FALSE) + 10 < humi.coretemperature) // with small buffer as temp can bounce a bit
```

Comment:
Why these two numbers? you said the 10 was a buffer, is this checking if the skin is close to the bodies temp? You mention sweat?

Reply:
If your body temp goes over your normal you get sweaty and the effectiveness of insulation goes down.
The value of +10 here allows the body temp to go 10 points higher then normal without triggering the reduction in insulation. updating documentation.